src/Altinn.App.Core/Internal/Expressions/LayoutEvaluator.cs

```diff
@@ -122,7 +122,16 @@ await HiddenFieldsForRemovalRecurs(
     /// <summary>
     /// Remove fields that are only referenced from hidden fields from the data object in the state.
     /// </summary>
-    public static async Task RemoveHiddenData(LayoutEvaluatorState state, RowRemovalOption rowRemovalOption)
+    [Obsolete("Use the async version of this method RemoveHiddenDataAsync")]
+    public static void RemoveHiddenData(LayoutEvaluatorState state, RowRemovalOption rowRemovalOption)
+    {
+        RemoveHiddenDataAsync(state, rowRemovalOption).Wait();
```

Comment:
`GetAwaiter().GetResult()` instead? I think behavior around exceptions is better

Reply:
Seems to be that `.Wait()` wraps exceptions in AggregateExceptions, while `.GetAwaiter().GetResult()` unwraps the exception just like `await task` does. 
```suggestion
        RemoveHiddenDataAsync(state, rowRemovalOption).GetAwaiter().GetResult();
```